### PR TITLE
208 violated arguments one of are not rendered correctly

### LINKED ISSUE #208: Violated arguments of `oneOf` are not rendered correctly
repro:

```java
import java.util.Arrays;

import am.ik.yavi.builder.ValidatorBuilder;
import am.ik.yavi.core.ConstraintViolations;
import am.ik.yavi.core.Validator;

public class Demo {
	public static void main(String[] args) {

		final Validator<Message> validator = ValidatorBuilder.<Message>of()
				.constraint(Message::getText, "text", c -> c.oneOf(Arrays.asList("a", "b")))
				._object(Message::getColor, "color", c -> c.oneOf(Arrays.asList(Color.RED, Color.BLUE)))
				.build();
		final ConstraintViolations violations = validator.validate(new Message("c", Color.GREEN));
		violations.forEach(v -> System.out.println(v.message()));
	}

	static class Message {
		private final String text;

		private final Color color;

		Message(String text, Color color) {
			this.text = text;
			this.color = color;
		}

		public String getText() {
			return text;
		}

		public Color getColor() {
			return color;
		}
	}

	enum Color {
		RED, BLUE, GREEN
	}
}

```

we'll see
```
"text" must be one of the following values: [Ljava.lang.Object;@327471b5
"color" must be one of the following values: [Ljava.lang.Object;@4157f54e
```

These messages should be
```
"text" must be one of the following values: [a, b]
"color" must be one of the following values: [RED, BLUE]
```

### PR DESCRIPTION
**Pull request by duponter**
Fixes #208

### CODE REVIEW

File: src/main/java/am/ik/yavi/core/Constraint.java [hunk 1]
```diff
@@ -51,8 +51,10 @@ default C equalTo(@Nullable V other) {
 	 * @since 0.10.0
 	 */
 	default C oneOf(Collection<V> values) {
-		this.predicates().add(ConstraintPredicate.of(values::contains, OBJECT_ONE_OF,
-				() -> new Object[] { values.toArray() }, NullAs.INVALID));
+		this.predicates()
+				.add(ConstraintPredicate.of(values::contains, OBJECT_ONE_OF,
+						() -> new Object[] { new ArrayArgument(values.toArray()) },
```
Comment (duponter) [inline src/main/java/am/ik/yavi/core/Constraint.java:56]: If you prefer a simpler solution, calling `Arrays.toString(values.toArray())` works as well. 
However, it is less flexible when you need the array later. 
In that case, you'll have to parse the toString output to recreate the array.

The advantage of this solution is that the original array can be retrieved without having the need of parsing the toString.

Comment (making) [reply src/main/java/am/ik/yavi/core/Constraint.java:56]: Maybe calling Arrays.asList is the simplest?

Comment (duponter) [reply src/main/java/am/ik/yavi/core/Constraint.java]: Even that is not necessary, it is sufficient to simply pass the incoming collection as argument. 
I changed the test a little bit to test oneOf both with a List and a Set.

Comment (making) [reply src/main/java/am/ik/yavi/core/Constraint.java]: thanks!

**Comment by making** [comment]
@ffroliva This fix will be included in 0.10.1, which is expected to be released soon.
I am planning to migrate the release process for the release of 0.10.1 with github action, which may take some time.
Until then, please use 0.10.1-SNAPSHOT.

**Comment by ffroliva** [comment]
Thanks for letting me know.

Regards,

Flavio Oliva

Em ter., 15 de fev. de 2022 às 03:28, Toshiaki Maki <
***@***.***> escreveu:

**Comment by making** [comment]
@ffroliva 0.10.1 has just been released.
https://github.com/making/yavi/releases/tag/0.10.1

**Comment by ffroliva** [comment]
Thank you!

Em qui., 17 de fev. de 2022 às 03:25, Toshiaki Maki <
***@***.***> escreveu:

**Pull request merged**